torch/csrc/jit/function.h

```diff
@@ -103,9 +103,13 @@ struct TORCH_API Function {
     if (executor_) {
       return executor_;
     }
+    // Make sure optimized graph is computed (do it before acquiring the lock to
```

Comment:
This is a bit tricky, we should either switch the mutex to one that can be nested acquired by the same thread, or separate the public private API so that we can ask for the optimized graph below the lock. Otherwise, this will get confusing.